### PR TITLE
[StarPU] Add builder

### PR DESCRIPTION
**Pull request by amontoison**

### CODE REVIEW

**Comment by amontoison** [comment]
I am able to build `StarPU` without error on my computer and it works for multiple linux platforms.

File: S/StarPU/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = expand_gfortran_versions(supported_platforms())
+
+
+# The products that we will ensure are always built
+products = [
+    LibraryProduct("libstarpu", :libstarpu)
```
Comment (giordano) [inline S/StarPU/build_tarballs.jl:32]: https://docs.binarybuilder.org/dev/reference/#BinaryBuilderBase.LibraryProduct
```julia
julia> using Base.BinaryPlatforms

julia> parse_dl_name_version("libstarpu-1.3.so.5", "linux")[1]
"libstarpu-1.3"
```
```suggestion
    LibraryProduct("libstarpu-1.3", :libstarpu)
```

File: S/StarPU/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+cd starpu*
+mkdir build
+cd build
+../configure --prefix=${prefix} --build=${MACHTYPE} --host=${target} --enable-fortran
+make -j${nproc}
+make install
+exit
```
Comment (giordano) [inline S/StarPU/build_tarballs.jl:22]:
```suggestion
install_license COPYING.LGPL
```

File: S/StarPU/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = expand_gfortran_versions(supported_platforms())
+
+
+# The products that we will ensure are always built
+products = [
+    LibraryProduct("libstarpu-1.3", :libstarpu),
```
Comment (giordano) [inline S/StarPU/build_tarballs.jl:32]: For macOS:
```julia
julia> using Base.BinaryPlatforms

julia> parse_dl_name_version("libstarpu-1.3.5.dylib", "macos")[1]
"libstarpu-1"
```
```suggestion
    LibraryProduct(["libstarpu-1.3", "libstarpu-1"], :libstarpu),
```

**Comment by amontoison** [comment]
@giordano Should I remove aarch64-linux-musl and Windows platforms ?

**Comment by giordano** [comment]
> @giordano Should I remove aarch64-linux-musl and Windows platforms ?

Up to you :slightly_smiling_face:

**Comment by amontoison** [comment]
I removed the Windows platforms.
For the error on `aarch64-linux-musl` do you have an idea how it can be solved or not ?
It's the only linux platform that doesn't passed.

**Comment by amontoison** [comment]
The build on the platform `aarch64-linux-musl` works on my computer and I can't reproduce the error on Yggdrasil builds. I removed this platform.

**Comment by giordano** [comment]
```
┌ Warning: share/doc/starpu/manual/html/files.html causes a case-sensitivity ambiguity!
└ @ BinaryBuilder.Auditor /depot/packages/BinaryBuilder/xoOuD/src/auditor/filesystems.jl:9
##[warning]share/doc/starpu/manual/html/files.html causes a case-sensitivity ambiguity!
```
It'd be good to either disable building documentation (if possible) or delete it.  If there are files with the same name module case, that's a recipe for a disaster on case-insensitive file systems

**Comment by amontoison** [comment]
I already added the option to not build the documentation : https://github.com/JuliaPackaging/Yggdrasil/pull/2681/files#diff-7c77425497603b157ef579b45cc6e081d678e006b5622b5a0c9a3b1fa993fd0dR19

We should delete it.

**Comment by giordano** [comment]
> I already added the option to not build the documentation

I like build systems that work :stuck_out_tongue:

**Comment by giordano** [comment]
https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=10171&view=logs&j=cce093cc-edbb-58e7-cbe4-331c7a363ed5&t=ffc73a8a-79a0-5708-dc97-b1095f94290f&l=5256
```
[16:06:12]         Documentation HTML:  no
[16:06:12]         Documentation PDF:   no
```
and yet they're both there...

**Comment by amontoison** [comment]
> ike build systems that work

Me too :stuck_out_tongue:

**Pull request merged**